uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/GuvnorM2Repository.java

```diff
@@ -460,22 +464,47 @@ public String getPomText(final String path) {
         }
     }
 
-    private static String loadPomFromJar(final File file) {
-        try (ZipFile zip = new ZipFile(file)) {
-            for (Enumeration e = zip.entries(); e.hasMoreElements(); ) {
-                ZipEntry entry = (ZipEntry) e.nextElement();
+    public String getKModuleText(final String path) {
+        return loadKieTextFromJar(path, META_INF, "kmodule.xml");
+    }
 
-                if (entry.getName().startsWith("META-INF/maven") && entry.getName().endsWith("pom.xml")) {
-                    return zipEntryToString(zip,
-                                            entry);
+    public String getKieDeploymentDescriptorText(final String path) {
+        return loadKieTextFromJar(path, META_INF, "kie-deployment-descriptor.xml");
```

Comment:
to be consistent we should introduce constant for kie-deployment-descriptor.xml and possibly also use it in the tests